code/modules/vehicles/droppod.dm

```diff
@@ -0,0 +1,223 @@
+
+	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "droppod.tmpl", "Drop Pod", 400, 400, state = state)
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/vehicle/droppod/Topic(href, href_list)
+	if(..())
+		return 1
+
+	if(href_list["fire"])
+		var/target = href_list["fire"]
+		if(target == "arrivals")
+			if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/fore)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/port)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/starboard)
+			else if(prob(20))
+				firefromarea(/area/hallway/secondary/entry/aft)
+			else
+				firefromarea(/area/maintenance/arrivals)
+		else if(target == "cargo")
+			if(prob(65))
+				firefromarea(/area/quartermaster/loading)
+			else
+				firefromarea(/area/quartermaster/qm)
+		else if(target == "tcoms")
+			if(prob(20))
+				firefromarea(/area/tcommsat/entrance)
+			else if(prob(5)) // could drop them on an openturf
+				firefromarea(/area/turret_protected/tcomsat)
+			else if(prob(20))
+				firefromarea(/area/turret_protected/tcomfoyer)
+			else if(prob(20))
+				firefromarea(/area/turret_protected/tcomwest)
+			else if(prob(20))
+				firefromarea(/area/turret_protected/tcomeast)
+			else if(prob(20))
+				firefromarea(/area/tcommsat/computer)
+			else
+				firefromarea(/area/tcommsat/powercontrol)
+		else if(target == "commandescape")
+			firefromarea(/area/bridge/levela)
+
+/obj/vehicle/droppod/proc/firefromarea(var/area/A)
+	if(A in protectedareas)
+		ermessage()
+		return
+
+	var/turf/targetturf = pick_area_turf(A)
+	if(iswall(targetturf))
+		ermessage(0,2)
+		firefromarea(A)
+	else
+		fire(targetturf)
+
+/obj/vehicle/droppod/proc/fire(var/turf/A)
+	if(!isturf(A))
+		ermessage()
+		return
+
+	if(!(src.z in validfirelocations))
+		ermessage(1, 1)
+		return
+
+	var/turf/aboveturf = GetAbove(A)
+	if(aboveturf)
+		if(aboveturf.is_hole)
+			A.visible_message("<span class='danger'>The [src] drops through the hole in the roof!</span>")
+			applyfalldamage(A)
+			forceMove(aboveturf)
+		else
+			applyfalldamage(aboveturf)
+			aboveturf.ChangeTurf(/turf/space)
+			applyfalldamage(A)
+			A.visible_message("<span class='danger'>The [src] crashes through the roof!</span>")
+			forceMove(A)
+
+		var/turf/belowturf = GetBelow(A)
+		if(belowturf)
+			//sound here
+			belowturf.visible_message("<span class='danger'>You hear something crash into the ceiling above!</span>")
+
+		if(humanload)
+			humanload.forceMove(loc)
+
+		if(passenger)
+			passenger.forceMove(loc)
+
+		used = 1
+
+/obj/vehicle/droppod/proc/applyfalldamage(var/turf/A)
+	for(var/mob/T in A)
+		T.gib()
+		T.visible_message("<span class='danger'>[T] is squished by the drop pod!</span>")
+	for(var/obj/B in A)
+		qdel(B)
+		B.visible_message("<span class='danger'>[B] is destroyed by the drop pod!</span>")
+
+/obj/vehicle/droppod/proc/ermessage(var/type = 0, var/subtype = 0)
```

Comment:
One `switch (type)` coupled with defines might be better? It's more CPU efficient basically that way.